Spigot-API-Patches/0219-NamespacedKey-based-plugin-chunk-tickets.patch

```diff
@@ -0,0 +1,280 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Trigary <trigary0@gmail.com>
+Date: Sat, 22 Aug 2020 14:41:43 +0200
+Subject: [PATCH] NamespacedKey based plugin chunk tickets
+
+
+diff --git a/src/main/java/io/papermc/paper/TickingLevel.java b/src/main/java/io/papermc/paper/TickingLevel.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..4eebf1c59d8047a48aca817a2547d2d2d840757e
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/TickingLevel.java
+@@ -0,0 +1,62 @@
++package io.papermc.paper;
++
++import org.bukkit.Chunk;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Represents what kind of ticking is done on the {@link Chunk}.
++ * The important values have dedicated cached instances, eg. {@link #ENTITY}.
++ * Read their own documentation for more information.
++ * <p>
++ * Ordering is based on the impact/importance of instances,
++ * so eg. {@link #ENTITY} is placed before {@link #BLOCK} when sorted.
++ * Traits defined by a specific level are inherited in the levels placed before it.
++ * <p>
++ * Instances of this class are not singletons even though there are some
++ * cached instances, eg. {@link #ENTITY}.
++ */
++public class TickingLevel implements Comparable<TickingLevel> {

```

Comment:
don't want to expose the magic values on API.

This class should be an enum and then server can map enum to magic value.

Don't want plugins adding stupid ticket levels that mojang can change the behavior of.